File: front/lib/api/src/models/external_id.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'external_id.freezed.dart';
part 'external_id.g.dart';

@freezed
class ExternalId with _$ExternalId {
  const factory ExternalId({
    required String url,
    required String value,
    required String? description,
    required int? rating,
    @JsonKey(name: "provider_name")
    required String providerName,
  }) = _ExternalId;

  factory ExternalId.fromJson(Map<String, dynamic> json) => _$ExternalIdFromJson(json);
}

File: front-vue/models/domain/external-id.ts
import * as yup from "yup";

const ExternalId = yup.object({
    url: yup.string().required(),
    value: yup.string().required(),
    description: yup.string().required().nullable(),
    rating: yup.number().required().nullable(),
    provider_name: yup.string().required(),
});

type ExternalId = yup.InferType<typeof ExternalId>;

export { ExternalId };
